File: JAVA-Stack/Java Fundamentals/FizzBuzz.java
// Write a program that will take an integer and print Fizz if the number is divisible by 3, Buzz if the number is divisible by 5, FizzBuzz if the number is divisible by 3 and 5, and the number itself for all other cases.

// Objectives:
// ● Practice conditionals.
// ● Implement a classic interview algorithm.


public class FizzBuzz {
    public String fizzBuzz(int num) {
        // fizzbuzz logic here
        if(num % 15 == 0 ){
            return "FizzBuzz";
        }
        else if(num % 3 ==0){
            return "Fizz";
        }
        else if(num % 5 ==0){
            return "Buzz";
        }
        else {
            return String.valueOf(num);
        }
    }
}

File: MEAN-Stack/Prework/FizzBuzz.js
function fizzbuzz(n) {
    // validate if n is a positive number
    if ( n < 0 ) {
        console.log("n has to be a positive number")
    }
    else if ( isNaN(n) ) {
        console.log("n is not number")
    } 
    else {
        for ( var i = 1; i <= n; i++ ) {
            if ( i % 3 == 0 && i % 5 == 0 ) {
                console.log("FizzBuzz")
            }
            else if ( i % 3 == 0 && i % 5 >= 1 ) {
               console.log("Fizz")
            }
            else if ( i % 5 == 0 && i % 3 >= 1  ) {
                console.log("Buzz")
            }
            else { 
                console.log(i) 
            }
        }
    }
}
    fizzbuzz("-15")